client/src/client.rs

```diff
@@ -360,55 +350,41 @@ impl<DB: Database> Client<DB> {
 
     #[cfg(target_arch = "wasm32")]
     fn start_advance_thread(&self) {
-        use std::time::Duration;
-        use tokio::sync::Mutex;
-
-        let node = self.node.clone();
-        let update_guard = Arc::new(Mutex::new(()));
+        let arc_swap_node = self.node.clone();
+        let updating = Arc::new(AtomicBool::new(false));
 
         let timer_id = set_timer_interval(Duration::from_secs(12), move || {
-            let node = node.clone();
-            let update_guard = update_guard.clone();
+            let arc_swap_node = arc_swap_node.clone();
+            let updating = updating.clone();
 
             spawn(async move {
-                let _guard = match update_guard.try_lock() {
-                    Ok(guard) => guard,
-                    _ => {
-                        log::debug!("Advancing already in progress");
-                        return;
-                    }
-                };
-
-                info!("Advancing the client");
+                if updating
+                    .compare_exchange(false, true, Ordering::SeqCst, Ordering::SeqCst)
```

Comment:
wouldn't `Relaxed, Relaxed` do the work there? or `Acquire, Relaxed`?
`wasm32` targets are single threaded anyway and `SeqCst` probably hurts compiler the most

Reply:
Since it is single-threaded environment, I changed `Arc<AtomicBool>` to `Rc<Cell<bool>>`